src/components/tx-flow/flows/RecoveryAttempt/RecoveryAttemptReview.tsx

```diff
@@ -0,0 +1,112 @@
+import { type SyntheticEvent, useState, useContext, useCallback, useReducer } from 'react'
+import { CircularProgress, CardActions, Button, Typography, Stack, Divider } from '@mui/material'
+import EthHashInfo from '@/components/common/EthHashInfo'
+import CheckWallet from '@/components/common/CheckWallet'
+import { Errors, trackError } from '@/services/exceptions'
+import { dispatchRecoveryExecution } from '@/features/recovery/services/recovery-sender'
+import useWallet from '@/hooks/wallets/useWallet'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import ErrorMessage from '@/components/tx/ErrorMessage'
+import TxCard from '@/components/tx-flow/common/TxCard'
+import FieldsGrid from '@/components/tx/FieldsGrid'
+import { TxModalContext } from '@/components/tx-flow'
+import NetworkWarning from '@/components/new-safe/create/NetworkWarning'
+import { RecoveryValidationErrors } from '@/features/recovery/components/RecoveryValidationErrors'
+import type { RecoveryQueueItem } from '@/features/recovery/services/recovery-state'
+
+type RecoveryAttemptReviewProps = {
+  item: RecoveryQueueItem
+}
+
+function useAsyncDispatch<T extends (...args: any) => Promise<any>>(dispatchFn: T) {
+  const [isLoading, toggleIsLoading] = useReducer((isLoading: boolean) => !isLoading, false)
+  const [error, setError] = useState<Error>()
+
+  const asyncDispatch = useCallback(
+    async (...args: Parameters<T>) => {
+      toggleIsLoading()
+      setError(undefined)
+
+      try {
+        const data = await dispatchFn(...args)
+
+        toggleIsLoading()
+
+        return data
+      } catch (e) {
+        trackError(Errors._812, e)
+        setError(e as Error)
+        toggleIsLoading()
+      }
+    },
+    [dispatchFn],
+  )
+
+  return { isLoading, error, asyncDispatch }
+}
+
+const RecoveryAttemptReview = ({ item }: RecoveryAttemptReviewProps) => {
+  const { asyncDispatch, isLoading, error } = useAsyncDispatch(dispatchRecoveryExecution)
+  const wallet = useWallet()
+  const { safe } = useSafeInfo()
+  const { setTxFlow } = useContext(TxModalContext)
+
+  const onFormSubmit = useCallback(
+    async (e: SyntheticEvent) => {
+      e.preventDefault()
+
+      if (!wallet) return
+
+      asyncDispatch({
+        provider: wallet.provider,
+        chainId: safe.chainId,
+        args: item.args,
+        delayModifierAddress: item.address,
+        signerAddress: wallet.address,
+      })
+    },
+    [asyncDispatch, wallet, safe, item.address, item.args],
+  )
+
+  return (
+    <TxCard>
+      <form onSubmit={onFormSubmit}>
+        <Stack gap={3} mb={2}>
+          {item.address && (
+            <FieldsGrid title="Initiated by">
+              <EthHashInfo address={item.address} showAvatar hasExplorer showName />
+            </FieldsGrid>
```

Comment:
I would expect to see the recoverer address here (the wallet that initiated this recovery attempt) but `item.address` is the contract address of the delay modifier.

Reply:
Oh, good catch.

Reply:
We could even include the validity time since its part of `item` i.e. say `Confirm or reject this transaction until 20.11.2024` or if it never expires i.e. `item.expiresAt` is null we could say `Confirm or reject this transaction`